pkgs/by-name/cc/cctools/package.nix

```diff
@@ -0,0 +1,142 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildPackages,
+  darwin,
+  ld64,
+  llvm,
+  memstreamHook,
+  meson,
+  ninja,
+  gitUpdater,
+}:
+
+let
+  # The targetPrefix is prepended to binary names to allow multiple binuntils on the PATH to both be usable.
+  targetPrefix = lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform) "${stdenv.targetPlatform.config}-";
+
+  # First version with all the required files
+  xnu = fetchFromGitHub {
+    name = "xnu-src";
+    owner = "apple-oss-distributions";
+    repo = "xnu";
+    rev = "xnu-7195.50.7.100.1";
+    hash = "sha256-uHmAOm6k9ZXWfyqHiDSpm+tZqUbERlr6rXSJ4xNACkM=";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "${targetPrefix}cctools";
+  version = "1010.6";
+
+  outputs = [
+    "out"
+    "dev"
+    "man"
+    "gas"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "apple-oss-distributions";
+    repo = "cctools";
+    rev = "cctools-${finalAttrs.version}";
+    hash = "sha256-JiKCP6U+xxR4mk4TXWv/mEo9Idg+QQqUYmB/EeRksCE=";
+  };
+
+  sourceRoot = "source";
+
+  postUnpack = ''
+    unpackFile '${xnu}'
+  '';
+
+  patches = [
+    # Fix compile errors in redo_prebinding.c
+    ./0001-Fix-build-issues-with-misc-redo_prebinding.c.patch
+    # Use libcd_is_blob_a_linker_signature as defined in the libcodedirectory.h header
+    ./0002-Rely-on-libcd_is_blob_a_linker_signature.patch
+    # cctools uses availability checks for `utimensat`, but it checks the wrong version.
+    # Also, provide a definition to avoid implicit function definition errors.
+    ./0003-Fix-utimensat-compatability-with-the-10.12-SDK.patch
+    # Use the nixpkgs clang’s path as the prefix.
+    ./0004-Use-nixpkgs-clang-with-the-assembler-driver.patch
```

Comment:
Thanks for the suggestion. I went with a variation of this patch. I left the `LOCALLIB` part out to keep the diff size down and assigned the prefixes as `@<package>@/bin/` since that’s what the code expects from the prefix.